prefect_sqlalchemy/credentials.py

```diff
@@ -253,3 +260,811 @@ def dict(self, *args, **kwargs) -> Dict:
+    close the engine and its connections when the context is exited.
+
+    It is also recommended that this block is loaded and consumed within a single task
+    or flow because if the block is passed across separate tasks and flows,
+    the state of the block's connection and cursor could be lost.
+
+    Attributes:
+        driver: The driver name, e.g. "postgresql+asyncpg"
+        database: The name of the database to use.
+        username: The user name used to authenticate.
+        password: The password used to authenticate.
+        host: The host address of the database.
+        port: The port to connect to the database.
+        query: A dictionary of string keys to string values to be passed to
+            the dialect and/or the DBAPI upon connect. To specify non-string
+            parameters to a Python DBAPI directly, use connect_args.
+        url: Manually create and provide a URL to create the engine,
+            this is useful for external dialects, e.g. Snowflake, because some
+            of the params, such as "warehouse", is not directly supported in
+            the vanilla `sqlalchemy.engine.URL.create` method; do not provide
+            this alongside with other URL params as it will raise a `ValueError`.
+        connect_args: The options which will be passed directly to the
+            DBAPI's connect() method as additional keyword arguments.
+        fetch_size: The number of rows to fetch at a time.
+
+    Example:
+        Load stored database credentials and use in context manager:
+        ```python
+        from prefect_sqlalchemy import SqlAlchemyConnector
+
+        database_block = SqlAlchemyConnector.load("BLOCK_NAME")
+        with database_block:
+            ...
+        ```
+
+        Create table named customers and insert values; then fetch the first 10 rows.
+        ```python
+        from prefect_sqlalchemy import SqlAlchemyConnector, SyncDriver
+
+        with SqlAlchemyConnector(
+            driver=SyncDriver.SQLITE_PYSQLITE,
+            database="prefect.db"
+        ) as database:
+            database.execute(
+                "CREATE TABLE IF NOT EXISTS customers (name varchar, address varchar);",
+            )
+            for i in range(1, 42):
+                database.execute(
+                    "INSERT INTO customers (name, address) VALUES (:name, :address);",
+                    parameters={"name": "Marvin", "address": f"Highway {i}"},
+                )
+            results = database.fetch_many(
+                "SELECT * FROM customers WHERE name = :name;",
+                parameters={"name": "Marvin"},
+                size=10
+            )
+        print(results)
+        ```
+    """
+
+    _block_type_name = "SQLAlchemy Connector"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/37TOcxeP9kfXffpKVRAHiJ/0f359112e79d0bd3dfe38c73c4fc6363/sqlalchemy.png?h=250"  # noqa
+
+    url: Union[SqlAlchemyUrl, AnyUrl] = Field(
```

Comment:
Not sure about this name, but `url` doesn't seem to quite mesh with the new model.
```suggestion
    connection_info: Union[SqlAlchemyUrl, AnyUrl] = Field(
```